segmentation/utils/spatial_analysis_utils_test.py

```diff
@@ -208,3 +217,35 @@ def test_calculate_enrichment_stats():
     assert abs(stats_xr.loc["z", 0, 0]) < 1
     assert stats_xr.loc["p_neg", 0, 0] > .05
     assert stats_xr.loc["p_pos", 0, 0] > .05
+
+
+def test_compute_neighbor_count():
+    fov_col = "SampleID"
+    flowsom_col = "FlowSOM_ID"
+    cell_label_col = "cellLabelInImage"
+    cell_type_col = "cell_type"
+    distlim = 100
+
+    fov_data, dist_matrix = make_example_data_closenum()
+
+    pheno_titles = fov_data[cell_type_col].drop_duplicates()
+    fov_data = fov_data[[fov_col, cell_label_col, flowsom_col, cell_type_col]]
+    pheno_num = len(fov_data[flowsom_col].drop_duplicates())
+
+    cell_neighbor_counts = pd.DataFrame(np.zeros((fov_data.shape[0], pheno_num + 2)))
+    cell_neighbor_freqs = cell_neighbor_counts.copy(deep=True)
+
+    cell_neighbor_counts[[0, 1]] = fov_data[[fov_col, cell_label_col]]
+    cell_neighbor_freqs[[0, 1]] = fov_data[[fov_col, cell_label_col]]
+
+    # Rename the columns to match cell phenotypes
+    cols = [fov_col, cell_label_col] + list(pheno_titles)
+    cell_neighbor_counts.columns = cols
+    cell_neighbor_freqs.columns = cols
+
```

Comment:
See earlier comment about moving `cell_neighbor_freqs` copy under this code block.